src/invoices/InvoiceForm/InvoiceForm.js

```diff
@@ -155,15 +155,17 @@ const InvoiceForm = ({
       const vendorAccountingCode = hasAnyAccountingCode ? null : erpCode;
       const accountNo = hasAnyAccountingCode ? null : NO_ACCOUNT_NUMBER;
       const exportToAccounting = Boolean(vendor.exportToAccounting);
+      const vendorPreferredCurrency = vendor?.vendorCurrencies?.slice(-1)[0];
```

Comment:
I don't think we need to iterate through the entire array just to retrieve the last element. (`.at(-1)`)